### PR TITLE
Package metadata

### PR DESCRIPTION
**Pull request by lispandfound**
Sets the git project so that it can be installed via `pip` and used from other projects. It is using pyproject.toml primarily, rather than the legacy setup.py. To maintain compatibility with legacy systems that might require a setup.py file, I have written a stub file that just bootstraps the pyproject.toml installation as per [the official setuptools package documentation](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html). The dependencies are dynamically sourced from requirements.txt. I think this splits the difference between:

1. Using a modern build-system for the package that is less brittle and easier to maintain.
2. Addressing @joelridden's concerns about using different package managers and virtual environment workflows by still maintaining a requirements.txt file.
3. Addressing @claudio525's concerns about mamba and miniforge installations by still supporting setup.py (but without duplicating work by maintaining both in parallel).

I will confirm and check that mamba will work with this setup. I do not anticipate problems with this because otherwise almost no pip package would work with mamba.

### CODE REVIEW

**Comment by lispandfound** [comment]
One snag here is that deptry doesn't like the fact that the requirements are dynamically sourced from a requirements.txt file, but the project is specified in a pyproject.toml file. Hence deptry will fail on any project with this configuration. I can make a PR to the deptry project to fix this.

**Comment by lispandfound** [comment]
There is now a PR in https://github.com/fpgmaas/deptry/pull/724, we may just be able to use my fork of deptry in the meantime for dependency checking.

File: setup.py
Comment (claudio525) [inline setup.py:1]: As mentioned at standup, this is not required for mamba support. Internal/local packages are just installed into mamba environments using `pip`.

**Review by sungeunbae — DISMISSED** [review]

File: setup.py
Comment (lispandfound) [reply setup.py:1]: ok then I'll remove it.

**Review by joelridden — APPROVED** [review]

**Review by AndrewRidden-Harper — APPROVED** [review]

**Pull request merged**